livre/eq/zeroeq/communicator.cpp

```diff
@@ -243,32 +246,20 @@ class Communicator::Impl
         if (!_httpServer)
             return;
 
-        _httpServer->handlePUT(::lexis::render::Exit::ZEROBUF_TYPE_NAME(),
-                               ::lexis::render::Exit::ZEROBUF_SCHEMA(),
-                               [&] { return requestExit(); });
-
-        _httpServer->handleGET(::lexis::render::ImageJPEG::ZEROBUF_TYPE_NAME(),
-                               ::lexis::render::ImageJPEG::ZEROBUF_SCHEMA(),
-                               [&] { return _config.renderJPEG(); });
-
-        _httpServer->handlePUT(::lexis::render::LookOut::ZEROBUF_TYPE_NAME(),
-                               ::lexis::render::LookOut::ZEROBUF_SCHEMA(),
-                               [&](const std::string& json) {
-                                   ::lexis::render::LookOut lookOut;
-                                   if (!lookOut.fromJSON(json))
-                                       return false;
-
-                                   onCamera(lookOut);
-                                   return true;
-                               });
-
-        _httpServer->handleGET(::lexis::render::LookOut::ZEROBUF_TYPE_NAME(),
-                               ::lexis::render::LookOut::ZEROBUF_SCHEMA(), [&] {
-                                   return _getLookOut(_getFrameData()
-                                                          .getCameraSettings()
-                                                          .getModelViewMatrix())
-                                       .toJSON();
-                               });
+        _exit.registerDeserializedCallback([this]() { requestExit(); });
+        _httpServer->handlePUT(_exit);
+
+        _imageJPEG.registerSerializeCallback(
+            [this]() { _config.renderJPEG(_imageJPEG); });
+        _httpServer->handleGET(_imageJPEG);
+
+        _lookOut.registerDeserializedCallback([this]() { onCamera(_lookOut); });
+        _lookOut.registerSerializeCallback([this]() {
+            const auto& camera = _getFrameData().getCameraSettings();
+            const auto lookout = _getLookOut(camera.getModelViewMatrix());
```

Comment:
Yes, I like to keep the type of the object explicit whenever possible. The first is a const ref to an *existing* CameraSettings object. The second is a *new* lookout object created by the function (which maybe should not have been called _getLookOut but _createLookOut). I find that binding const refs to temporaries obscures what is really happening.